etc/pdg_table_update.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/python
+
+import urllib.request
+import sys
+
+# Character positions for extracting data from "mass_width_2023.txt"
+MASS_WIDTH_ID_START = 0
+MASS_WIDTH_ID_END = 8
+MASS_WIDTH_MASS_START = 33
+MASS_WIDTH_MASS_END = 51
+MASS_WIDTH_WIDTH_START = 70
+MASS_WIDTH_WIDTH_END = 88
+MASS_WIDTH_NAME_WITH_CHARGE_START = 107
+MASS_WIDTH_NAME_WITH_CHARGE_END = 128
+
+#------------------------
+# Gets the text document
+#------------------------
+
+def getFileTextFromURL(link):
+    try:
+        with urllib.request.urlopen(link) as f:
+            text = f.read().decode("utf-8")
+            lines = text.splitlines()
+            return lines
+    except Exception as e:
+        print(f"Error while fetching data from URL: {e}")
+        return None
+
+#-------------------------------------
+# Gets the correct Data for particles
+#-------------------------------------
+
+def fixedParticleData(infoLine):
+    elements = infoLine.split()
+    
+    name = elements[1]
+    trueId = int(elements[2])
+
+    if len(elements) < 6:  # If it's an Antiparticle
+        return f"{elements[0]:>5} {name:<16} {trueId:>6} {elements[3]:>7} {elements[4]:>5}\n"
+    else:  # If it's not an Antiparticle
+        mass = float(elements[7])
+        width = float(elements[8])
+            
+        if trueId in massWidthData:
+            if massWidthData[trueId]['Mass']:
+                mass = float(massWidthData[trueId]['Mass'])
+            if massWidthData[trueId]['Width']:
+                width = float(massWidthData[trueId]['Width'])
+                
+        # Formatting the output line with the updated mass and width
+        return f"{elements[0]:>5} {name:<16} {trueId:>6} {elements[3]:>2} {elements[4]:>3} {elements[5]:<10} {elements[6]:>2} {mass:<11e} {width:<11e} {elements[9]:>3} {elements[10]:>2} {elements[11]:>3} {elements[12]:>2} {elements[13]:>4} {elements[14]:>3}\n"
```

Comment:
This seems to simply print the old data, not the updated one? It seems it simply reproduces the original values, with slightly changed format.

You can see that e.g. `eta_c` has the old mass value (2.980300e+00) not the one from `mass_width_2023.txt` which would be 2.9839E+00, i.e. old is 2.9803 but should now be 2.9839.

Or did I misunderstand the code?

Reply:
Thanks for the feedback! I'll fix the bug right away!